mono/metadata/image.c

```diff
@@ -719,6 +700,13 @@ mono_image_load_module_checked (MonoImage *image, int idx, MonoError *error)
 	if (image->modules_loaded [idx - 1])
 		return image->modules [idx - 1];
 
+#ifdef ENABLE_NETCORE
+	/* SRE still uses image->modules, but they are not loaded from files, so the rest of this function is dead code for netcore */
+	g_assert_not_reached ();
+#else
+
+	mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_ASSEMBLY, "Loading module %d of %s (%s)", idx, image->assembly ? image->assembly->aname.name : "some assembly", image->name);
```

Comment:
What's the scenario where the assembly doesn't exist?

Reply:
Someone using the embedding API directly to only open images, not assemblies.